src/main/java/org/jvnet/hudson/test/recipes/LocalData.java

```diff
@@ -102,7 +102,7 @@ public void setup(JenkinsRule jenkinsRule, LocalData recipe) throws Exception {
             Method testMethod;
 
             try {
-                testMethod = desc.getTestClass().getDeclaredMethod(desc.getMethodName());
```

Comment:
This will not work for JUnit5 tests that do not have a JenkinsRule as parameter (which is a valid option) and are not public (which is also a valid option). 

One remedy would be to use both `getMethod` and `getDeclaredMethod`.

Reply:
> that do not have a JenkinsRule as parameter (which is a valid option)

How is that valid? `@LocalData` is expected to work within a JenkinsRule instance, right?

Reply:
Yes it is, however you can do something like this:

```Java
@WithJenkins
class TestClass {

  private JenkinsRule rule;

  @BeforeEach
  void before(JenkinsRule rule) {
    this.rule = rule;
  }
  
  @LocalData
  @Test
  void testMethod() {
   // do stuff with rule
  }
}
```